Decode/PerfItemValue.cs

```diff
@@ -603,16 +622,22 @@ public string GetString()
             PerfConvertOptions convertOptions = PerfConvertOptions.Default)
         {
             Debug.Assert(this.Metadata.TypeSize <= this.Bytes.Length);
-            Text.Encoding? encodingFromBom;
-            switch (this.Metadata.Encoding)
+            var baseEncoding = this.Metadata.Encoding;
+            switch (baseEncoding)
             {
                 default:
-                    throw new NotSupportedException("Unknown encoding.");
-                case EventHeaderFieldEncoding.Struct:
-                    throw new InvalidOperationException("Invalid encoding for AppendScalarTo.");
+                    sb.Append("Encoding[");
+                    PerfConvert.UInt32DecimalAppend(sb, (uint)baseEncoding);
+                    sb.Append(']');
+                    break;
```

Comment:
Note: Instead of throwing for edge cases, just add an error message to the output string.